traversal/predicate/PredicateOperator.java

```diff
@@ -85,9 +86,13 @@ public Equality(TypeQLToken.Predicate.Equality token) {
             super(token);
         }
 
-        abstract boolean apply(int comparisonResult);
+        public static Equality of(TypeQLToken.Predicate.Equality operator) {
+            return Equality.operators.get(operator);
+        }
 
-        abstract Equality reflection();
+        public abstract boolean apply(int comparisonResult);
+
+        abstract public Equality reflection();
```

Comment:
Reflection doesn't need to be public, but apply does.
Should I revert this to package-private?